index.js

```diff
@@ -75,6 +77,35 @@ SubresourceIntegrityPlugin.prototype.apply = function apply(compiler) {
     }).join(' ');
   }
 
+  /* html-webpack-plugin has added an event so we can pre-process the html tags before they
+   inject them. This does the work.
+   */
+  function processTag(tag, compilation) {
+    if (!tag.attributes) return;
+    var src =  tag.attributes.href; // link tags have href
+    if (!src) {
+      src = tag.attributes.src; // script tags have src
+    }
+    if (!src) {
+      return;
+    }
+    var base = path.basename(src);
+    var asset = compilation.assets[base];
+    if (asset && asset.integrity) {
+      tag.attributes.integrity = asset.integrity;
+      tag.attributes.crossorigin = 'anonymous';
+    }
```

Comment:
I think this should be followed by something like

```
    } else {
      compilation.warnings.push(new Error(
        "webpack-subresource-integrity: cannot determine hash for asset '" +
          base + "', the resource will be unprotected."));
    }
```

Or perhaps even push to `compilation.errors` considering that all assets should come with a hash.